modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/ImageMosaicConfigHandler.java

```diff
@@ -161,6 +158,8 @@
 
     private boolean useExistingSchema;
 
+    private boolean isCog;
```

Comment:
Recommend calling this one just "cog", "isCog" is the name of a getter method, should not be be used for a field.
Also in the property file, there is nothing else starting with "is", having a ``cog=true`` makes things probably easier.

Config wise, wondering if having a cog=true could come with some defaults, like for example:

```
CogRangeReader=it.geosolutions.imageioimpl.plugins.cog.HTTPRangeReader
Heterogeneous=false
cog=true
SuggestedSPI=it.geosolutions.imageioimpl.plugins.cog.CogImageReaderSpi
```
If those two are the common range reader and SPI, I would just assume them if not specified, and force configuration only if extra behavior is needed, e.g., trying to access a COG is a secured S3 bucket, for example.

Reply:
Right.
At the bottom of the code it is actually working like that more or less. When config is missing, defaults are used.

Reply:
Would be interesting to verify the defaults are working as intended then.

Reply:
Will add a test taking the indexer and stripping out those entries to double check

Reply:
Done